### ISSUE TITLE
Add Numpy 2.0 support

### ISSUE THREAD

**Issue by MichaelClerx** [CMA-ES/pycma#253]
Numpy 2 was just released, which has a few backwards incompatible changes.

https://numpy.org/devdocs/release/2.0.0-notes.html
https://numpy.org/devdocs/numpy_2_0_migration_guide.html

Including
- `np.array(x, copy=False)` now raises an exception if it has to copy, so should use `np.asarray(x)` instead.
- `np.Inf` can no longer be used, should be `np.inf` instead
- `np.NaN` must now be `np.nan`
- deprecated `alltrue` has been removed, should just be `all` now

Also a few tests fail because numpy now ~seems to maintain numpy data types when casting to a list~ indicates the numpy data type when using `repr`, which is used in the doctests: https://github.com/numpy/numpy/issues/26730

**Comment by MichaelClerx** [comment]
With the first four points fixed, the tests pass except for some `repr` issues e.g.

```
File "/home/michael/dev/ext/pycma/cma/bbobbenchmarks.py", line 72, in bbobbenchmarks.py
Failed example:
    f3([0, 1, 2]) # short-cut for f3.evaluate([0, 1, 2]) # doctest:+ELLIPSIS
Expected:
    59.8733529...
Got:
    np.float64(59.87335290959168)
```

Suggested workaround is to configure "legacy printing" mode: https://github.com/numpy/numpy/issues/26730#issuecomment-2174256528

**Comment by MichaelClerx** [comment]
Just saw it's fixed in development 🙈

**Comment by MichaelClerx** [comment]
Are there plans for a new release soon?
We test a wrapper around `pycma` which is currently failing tests due to these issues https://github.com/pints-team/pints/pull/1531

**Comment by nikohansen** [comment]
Thanks for the comment and sorry for the late reply. Yes, there will be a release in 2-3 weeks.

**Comment by nikohansen** [comment]
> Are there plans for a new release soon? We test a wrapper around `pycma` which is currently failing tests due to these issues [pints-team/pints#1531](https://github.com/pints-team/pints/pull/1531)

`cma` version `3.4.0` was just released and should be compatible with `numpy 2.0`.
